src/Compilers/Core/Portable/Serialization/ObjectWriter.cs

```diff
@@ -670,13 +680,15 @@ private void WriteKnownType(Type type)
             this.WriteInt32(_binderSnapshot.GetTypeId(type));
         }
 
-        private void WriteObject(object instance)
+        private void WriteObject(object instance, IObjectWritable instanceAsWritableOpt)
```

Comment:
:question: Does it help to make a generic form on `T : IObjectWritable`? It could be used to turn the interface dispatch into a virtual dispatch. The `WriteObjectWorker` method would need to be made generic in the same fashion.